src/ImageCaptureCore/Defs.cs

```diff
@@ -0,0 +1,287 @@
+//
+
+//
+// Defs.cs: Enumerations and definitions for ImageCaptureCore
+//
+using System;
+
+using Foundation;
+using ObjCRuntime;
+
+namespace ImageCaptureCore {
+
+	[Native]
+	public enum ICExifOrientationType : ulong {
+		Orientation1 = 1,
+		Orientation2 = 2,
+		Orientation3 = 3,
+		Orientation4 = 4,
+		Orientation5 = 5,
+		Orientation6 = 6,
+		Orientation7 = 7,
+		Orientation8 = 8,
+	}
+
+	public enum ICReturnCodeOffset {
+		Thumbnail = -21000,
+		Metadata = -21050,
+		Download = -21100,
+		Delete = -21150,
+		ExFat = -21200,
+		Ptp = -21250,
+		System = -21300,
+	}
+
+	[Native]
+	public enum ICReturnCode : long {
+		Success = 0,
+		InvalidParam = -9922,
+		CommunicationTimedOut = -9923,
+		ScanOperationCanceled = -9924,
+		ScannerInUseByLocalUser = -9925,
+		ScannerInUseByRemoteUser = -9926,
+		DeviceFailedToOpenSession = -9927,
+		DeviceFailedToCloseSession = -9928,
+		ScannerFailedToSelectFunctionalUnit = -9929,
+		ScannerFailedToCompleteOverviewScan = -9930,
+		ScannerFailedToCompleteScan = -9931,
+		ReceivedUnsolicitedScannerStatusInfo = -9932,
+		ReceivedUnsolicitedScannerErrorInfo = -9933,
+		DownloadFailed = -9934,
+		UploadFailed = -9935,
+		FailedToCompletePassThroughCommand = -9936,
+		DownloadCanceled = -9937,
+		FailedToEnableTethering = -9938,
+		FailedToDisableTethering = -9939,
+		FailedToCompleteSendMessageRequest = -9940,
+		DeleteFilesFailed = -9941,
+		DeleteFilesCanceled = -9942,
+		DeviceIsPasscodeLocked = -9943,
+		DeviceFailedToTakePicture = -9944,
+		DeviceSoftwareNotInstalled = -9945,
+		DeviceSoftwareIsBeingInstalled = -9946,
+		DeviceSoftwareInstallationCompleted = -9947,
+		DeviceSoftwareInstallationCanceled = -9948,
+		DeviceSoftwareInstallationFailed = -9949,
+		DeviceSoftwareNotAvailable = -9950,
+		DeviceCouldNotPair = -9951,
+		DeviceCouldNotUnpair = -9952,
+		DeviceNeedsCredentials = -9953,
+		DeviceIsBusyEnumerating = -9954,
+		DeviceCommandGeneralFailure = -9955,
+		DeviceFailedToCompleteTransfer = -9956,
+		DeviceFailedToSendData = -9957,
+		SessionNotOpened = -9958,
+		ThumbnailNotAvailable = ICReturnCodeOffset.Thumbnail,
+		ThumbnailAlreadyFetching = ICReturnCodeOffset.Thumbnail - 1,
+		ThumbnailCanceled = ICReturnCodeOffset.Thumbnail - 2,
+		ThumbnailInvalid = ICReturnCodeOffset.Thumbnail - 3,
+		ErrorDeviceEjected = ICReturnCodeOffset.System,
+		MetadataNotAvailable = ICReturnCodeOffset.Metadata,
+		MetadataAlreadyFetching = ICReturnCodeOffset.Metadata - 1,
+		MetadataCanceled = ICReturnCodeOffset.Metadata - 2,
+		MetadataInvalid = ICReturnCodeOffset.Metadata - 3,
+		MultiErrorDictionary = -30000,
+	}
+
+	[Native]
+	public enum ICDeviceType : ulong {
+		Camera = 1,
+		Scanner = 2,
+	}
+
+	[Native]
+	public enum ICDeviceLocationType : ulong {
+		Local = 256,
+		Shared = 512,
+		Bonjour = 1024,
+		Bluetooth = 2048,
+	}
+
+	// Combination of ICDeviceTypeMask and ICDeviceLocationTypeMask
+	[Native]
+	[Flags]
+	public enum ICBrowsedDeviceType : ulong {
```

Comment:
add `None = 0,`
https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1008-enums-should-have-zero-value?view=vs-2017